katprep/snapshot.py

```diff
@@ -57,240 +51,272 @@ def parse_options(args=None):
-
-    try:
-        with open(OUTPUT_FILE, 'w') as target:
-            target.write(json.dumps(SYSTEM_ERRATA))
-    except IOError as err:
-        LOGGER.error("Unable to store report: '%s'", err)
+            with open(OUTPUT_FILE, "w") as target:
+                target.write(json.dumps(system_info))
+        except IOError as err:
+            LOGGER.error("Unable to store report: '%s'", err)
+        else:
+            LOGGER.info("Report '%s' created.", OUTPUT_FILE)
     else:
-        LOGGER.info("Report '%s' created.", OUTPUT_FILE)
-
+        LOGGER.info("Empty report - please check.")
 
 
 def main(options, args):
-    """Main function, starts the logic based on parameters."""
-    global SAT_CLIENT, OUTPUT_FILE
+    """
+    Main function, starts the logic based on parameters
+    """
+    global MGMT_CLIENT, OUTPUT_FILE
 
     LOGGER.debug("Options: %s", options)
     LOGGER.debug("Arguments: %s", args)
 
-    #set output file
+    # set output file
     if options.output_path == "":
         options.output_path = "./"
-    elif options.output_path != "" and \
-    options.output_path[len(options.output_path)-1:] != "/":
-        #add trailing slash
+    elif (
+            options.output_path != ""
+            and options.output_path[len(options.output_path) - 1:] != "/"
+    ):
+        # add trailing slash
         options.output_path = "{}/".format(options.output_path)
     OUTPUT_FILE = "{}errata-snapshot-report-{}-{}.json".format(
         options.output_path,
-        options.server.split('.')[0],
+        options.server.split(".")[0],
         time.strftime("%Y%m%d-%H%M")
     )
     LOGGER.debug("Output file will be: '%s'", OUTPUT_FILE)
 
-    #check if we can read and write before digging
+    # check if we can read and write before digging
     if is_writable(OUTPUT_FILE):
-        #initalize Foreman connection and scan systems
-        (sat_user, sat_pass) = get_credentials(
-            "Foreman", options.server, options.auth_container,
+        # initalize Foreman connection and scan systems
+        (mgmt_user, mgmt_pass) = get_credentials(
+            "Management",
+            options.server,
+            options.auth_container,
             options.auth_password
         )
-        SAT_CLIENT = ForemanAPIClient(
-            LOG_LEVEL, options.server, sat_user,
-            sat_pass, options.ssl_verify
+        MGMT_CLIENT = get_management_client(
+            options.mgmt_type,
+            LOG_LEVEL,
+            mgmt_user,
+            mgmt_pass,
+            options.server,
+            skip_ssl=options.skip_ssl,
         )
 
-        #validate filters
-        validate_filters(options, SAT_CLIENT)
+        # TODO: validate filters
+        # validate_filters(options, MGMT_CLIENT)
 
-        #scan systems and create report
-        scan_systems(options)
-        create_report()
+        # scan systems and create report
+        _info = scan_systems(options)
+        create_report(_info)
```

Comment:
I think the preceding underscore is not necessary. What is it for?

```suggestion
        info = scan_systems(options)
        create_report(info)
```